src/Core/Vipr/FileWriter.cs

```diff
@@ -26,8 +31,37 @@ public static void Write(IEnumerable<TextFile> textFilesToWrite, string outputDi
                 if (!Directory.Exists(Path.GetDirectoryName(filePath)))

```

Comment:
If you do add a {lock} you could put the directory creation inside that as well which guard against this issue.